API.md

```diff
@@ -0,0 +1,66 @@
+
+### `Octofet(uint8_t pinCS, uint8_t pinMOSI, uint8_t pinSCK, uint8_t devicesCount = 1)`
+
+Constructs a new object of the data bus software SPI with arguments:
+
+- `pinCS`: pin Chip Select is used to control the shift-register latch. Any GPIO-pin can be used.
+- `pinMOSI`: pin MOSI is used to send the data to the peripherals. Any GPIO-pin can be used.
+- `pinSCK`: pin SCK is used for the clock pulses which synchronize data transmission generated by the master. Any GPIO-pin can be used.
+- `devicesCount`: quantity of the Octofet boards connected in a chain: the logical output pin of one module is connected to the input of another. If the `devicesCount` argument is omitted, it controls only one Octofet board.
+
+The difference between the hardware and the software SPI is in transmission speed and availability. The software SPI will run much slower than the hardware SPI, but the hardware SPI works only with specific pins: `MOSI`, `MISO`(not used in Octofet) and `SCK`. Use the software SPI only when the hardware SPI pins are already used by other devices.
+
+### `void begin(const uint8_t* resetValues = NULL)`
+
+Initializes the given interface and prepares the board for communication. The argument `resetValues` sets the values of power switches of all devices by default as 8-bit. One bit for each power switch: `1` is "on" and `0` is "off". If the `resetValues` argument is adjusted, then when the `reset()` method is called, the power switches are set by the `resetValues`. If the `resetValues` argument is omitted, the power switches are set to "off" state.
+
+Important: call this method at least once, in your `setup()`.
+
+### `void digitalPinWrite(uint8_t pin, bool value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") of one power switch. Function arguments:
+
+- `pin`: the ordinal number of the power switch on the board. It ranges from `0` to `7`. Use an optional value `ALL` to select and set the state of all power switches simultaneously.
+- `value`: selects the state of the power switch. Values "on" and "off" may be put in different data types: `HIGH` or `LOW` / `true` or `false` / `1` or `0`.
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. Use an optional value `ALL` to select and set the state of all power switches simultaneously. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `void digitalPortWrite(uint8_t value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") for all power switches. Function arguments:
+
+- `value`: 8-bit value for all 8 power switches. One bit for each power switch: `1` is "on" and `0` is "off".
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. Use an optional value `ALL` to select and set the state of all devices. If the `device` argument is omitted, it controls only one Octofet board.
```

Comment:
I find this pin vs port variation is confusing. I’d better stay away from custom terms and concepts invented for Octofet. What about:

```cpp
void digitalWrite(uint8_t pin, bool value, uint8_t device = 0);
void digitalWrite8(uint8_t state, uint8_t device = 0);
```

Reply:
I suggest these names of methods:

```cpp
void digitalWrite(uint8_t switch, bool value, uint8_t device = 0);
void digitalWriteDevice(uint8_t value, uint8_t device = 0);
```